### ISSUE TITLE
Add a config to disable generating PR title and body from commit

### ISSUE THREAD

**Issue by krlvi** [gitbutlerapp/gitbutler#2870]
We want to have a config to disable https://github.com/gitbutlerapp/gitbutler/pull/2835

Discord thread: https://discord.com/channels/1060193121130000425/1210171236630659072/1210208452002058291

**Comment by mtsgrd** [comment]
Let's say you have this feature turned on, should we rather take the PR title from the branch title instead of taking from the commit? I've used this a few times now and it doesn't feel quite right as it is now.

**Comment by Pranav2612000** [comment]
I'll like to pick this up. Where do we want to add the config?

**Comment by Pranav2612000** [comment]
> Let's say you have this feature turned on, should we rather take the PR title from the branch title instead of taking from the commit? I've used this a few times now and it doesn't feel quite right as it is now.

Even if the PR has just 1 commit? The current behaviour makes sense

**Comment by mtsgrd** [comment]
Right, the reason I am questioning the PR title is because in GitButler you likely already have a title for the branch, which you don't have when creating them from branches using the CLI or in GitHub. What do  you think?

About the location, this should likely go under under "git stuff" in `gitbutler-ui/src/routes/settings/+page.svelte`.

Let me know if you need any help? I'll assign this ticket to you. :)

![image](https://github.com/gitbutlerapp/gitbutler/assets/1300356/470f438a-05d3-4029-b539-0b3854832fc0)

**Comment by Pranav2612000** [comment]
> Right, the reason I am questioning the PR title is because in GitButler you likely already have a title for the branch, which you don't have when creating them from branches using the CLI or in GitHub.
Yeah. It makes sense. If we already have a title for the branch, we can re use the same. How do you set the title for a branch ( I'm sorry for the dumb question, but I wasn't able to see a button anywhere )

**Comment by Pranav2612000** [comment]
Is something like this what you had in mind?
<img width="1160" alt="image" src="https://github.com/gitbutlerapp/gitbutler/assets/20909078/508b5b91-9145-4a2c-9584-52352eec52ee">

Things I need help with
- Finalizing the algorithm to use. Not sure what's the best way to discuss this and get the opinion of a wider spectrum of users.
- Text(Copy) for the config title
- Text(Copy) for the config description
- Do we need a `Learn More` like the above setting

Finalizing the algorithm is the most important part in my opinion. Once we have finalised it I can work up some proposals for the Config Title and Description ( If you have no specific standards it should stick to ) and we can iterate over them.

**Comment by Pranav2612000** [comment]
@mtsgrd Bumping this up^^ 😅
